code.js

```diff
@@ -0,0 +1,56 @@
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+
+// Initialize the S3 client outside the handler for reuse
+const s3Client = new S3Client();
+
+/**
+ * Lambda handler for processing orders and storing receipts in S3.
+ * @param {Object} event - Input event containing order details
+ * @param {string} event.order_id - The unique identifier for the order
+ * @param {number} event.amount - The order amount
+ * @param {string} event.item - The item purchased
+ * @returns {Promise<string>} Success message
+ */
+export const handler = async(event) => {
+    try {
+        // Access environment variables
+        const bucketName = process.env.RECEIPT_BUCKET;
+        if (!bucketName) {
+            throw new Error('RECEIPT_BUCKET environment variable is not set');
+        }
+
+        // Create the receipt content and key destination
+        const receiptContent = `OrderID: ${event.order_id}\nAmount: $${event.amount.toFixed(2)}\nItem: ${event.item}`;
+        const key = `receipts/${event.order_id}.txt`;
+
+        // Upload the receipt to S3
+        await uploadReceiptToS3(bucketName, key, receiptContent);
+
+        console.log(`Successfully processed order ${event.order_id} and stored receipt in S3 bucket ${bucketName}`);
+        return 'Success';
+    } catch (error) {
+        console.error(`Failed to process order: ${error.message}`);
+        throw error;
+    }
+};
+
```

Comment:
> [!WARNING]
> The error handling in the main handler wraps and rethrows the error, which could expose sensitive implementation details in the error message when the Lambda fails.

> [!TIP]
> Improve error handling to provide a cleaner error message without exposing internal details.

```javascript
try {
    // existing code
    return 'Success';
} catch (error) {
    console.error(`Failed to process order: ${error.message}`, { error, event });
    // Return a standardized error response instead of rethrowing
    return {
        statusCode: 500,
        body: JSON.stringify({ message: 'Failed to process order' })
    };
}
```